src/proto/media/file.rs

```diff
@@ -158,23 +243,127 @@ impl<'a> File<'a> {
     /// * `uefi::Status::ACCESS_DENIED`      The file was opened read only
     /// * `uefi::Status::VOLUME_FULL`        The volume is full
     pub fn flush(&mut self) -> Result<()> {
-        (self.inner.flush)(self.inner).into()
+        (self.0.flush)(self.0).into()
     }
 }
 
 impl<'a> Drop for File<'a> {
     fn drop(&mut self) {
-        let result: Result<()> = (self.inner.close)(self.inner).into();
+        let result: Result<()> = (self.0.close)(self.0).into();
         // The spec says this always succeeds.
         result.expect_success("Failed to close file");
     }
 }
 
+/// File wrapper for handling directories
+///
+/// The File abstraction can handle directories, but does so in a very roundabout way. A dedicated
+/// abstraction for directory handling that wraps the unintuitive File API is therefore desirable.
+pub struct Directory<'a>(File<'a>);
+
```

Comment:
Directory handling via File is a bit of a pain, so I added a dedicated abstraction to make it easier. Sadly, UEFI is designed such that querying if a file is a directory requires an unbounded amount of memory, so I cannot autodetect it for the user without memory allocation.